applications/ContactStructuralMechanicsApplication/custom_conditions/ALM_frictional_mortar_contact_condition.cpp

```diff
@@ -138,12 +138,13 @@ void AugmentedLagrangianMethodFrictionalMortarContactCondition<TDim,TNumNodes,TN
         const bool dual_LM = DerivativesUtilitiesType::CalculateAeAndDeltaAe(slave_geometry, normal_slave, master_geometry, rDerivativeData, rVariables, consider_normal_variation, conditions_points_slave, this_integration_method, this->GetAxisymmetricCoefficient(rVariables));
 
         for (IndexType i_geom = 0; i_geom < conditions_points_slave.size(); ++i_geom) {
-            std::vector<PointType::Pointer> points_array (TDim); // The points are stored as local coordinates, we calculate the global coordinates of this points
+            PointerVector< PointType > points_array; // The points are stored as local coordinates, we calculate the global coordinates of this points
+            points_array.reserve(TDim);
             array_1d<BelongType, TDim> belong_array;
             for (IndexType i_node = 0; i_node < TDim; ++i_node) {
                 PointType global_point;
                 slave_geometry.GlobalCoordinates(global_point, conditions_points_slave[i_geom][i_node]);
-                points_array[i_node] = Kratos::make_shared<PointType>(PointType(global_point));
+                points_array.push_back(Kratos::make_shared<PointType>(PointType(global_point)));
```

Comment:
avoid using push_back if you can.

here you know the size, hence define it as 

      PointerVector< PointType > points_array(TDim);

and then do as before but with the ()

      points_array(i_node) = Kratos::make_shared<PointType>(PointType(global_point));

Reply:
Ok

Reply:
Before i was using [], but it didn't compile, now I see that the operator [] is not the same as (). It is this OK?